src/protocol/sort/shuffle.rs

```diff
@@ -0,0 +1,100 @@
+use crate::helpers::prss::PrssSpace;
+use rand::rngs::mock::StepRng;
+use rand::seq::SliceRandom;
+
+#[allow(dead_code)]
+pub struct Shuffle {}
+
+impl Shuffle {
+    /// This implements Fisher Yates shuffle described here <https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle>
+    #[allow(clippy::cast_possible_truncation, dead_code)]
+    fn generate_random_permutation(
+        batchsize: usize,
+        index: u64,
+        left_permute: bool,
```

Comment:
I left this comment in another diff, but I should have left it here. I don't like using a boolean for this, because when you're reading the call site you don't know what the "true" or "false" you're seeing means. Also, the name "left_permute" means nothing.

I'd prefer an enum argument with only two options, like: "LEFT_SHARED_RANDOMNESS" and "RIGHT_SHARED_RANDOMNESS"

Or perhaps make two functions with slightly different names, like:

generate_random_permutation_left_randomness()
generate_random_permutation_right_randomness()

Reply:
I actually made the changes in the other diff to use your 1st suggestion of enum. We have an enum named Direction which is Left or Right. I have used that instead. Link : https://github.com/martinthomson/raw-ipa/pull/98/files#diff-894d6d54e26cbe3155f95c21f6bd6b557778f2705f8371fc2bc1d31b2292cb74R45
let me incorporate this in the relevant code here